nucypher/network/nodes.py

```diff
@@ -833,15 +837,18 @@ def learn_from_teacher_node(self, eager=False, canceller=None):
         try:
             metadata = MetadataResponse.from_bytes(response.content)
```

Comment:
Should we use `response.text` instead of `response.contest` as well?

Same question for lines 838, 841 and 850.

Reply:
The response code if we get here is 200 i.e. successful execution and therefore we only want the bytes that are returned.

Basically, on success (status code 200), use `response.content` for the bytes returned (serialized representation of various successful results eg. node metadata, reencrypted data etc.) from the endpoint. Those bytes then get deserialized to their corresponding objects such as `MetadataResponse` in this case.  It's not guaranteed that these bytes can be converted to a string for logging purposes.

Only on failure, we use the text because the data for the failure is expected to be a regular text string so calling `response.text` is used.

Reply:
It does feel like we should not be logging bytes though. Perhaps I'll convert those bytes to hex.

Reply:
Got it, thank you Derek!